Sources/Optimizely+Decide/OptimizelyUserContext.swift

```diff
@@ -119,18 +132,126 @@ public class OptimizelyUserContext {
     public func trackEvent(eventKey: String,
                            eventTags: OptimizelyEventTags? = nil) throws {
         
-        guard let optimizely = self.optimizely, optimizely.config != nil else {
+        guard let optimizely = self.optimizely else {
             throw OptimizelyError.sdkNotReady
         }
-
+        
         try optimizely.track(eventKey: eventKey,
                              userId: userId,
                              attributes: attributes,
                              eventTags: eventTags)
     }
+    
+}
 
+// MARK: - ForcedDecisions
+
+extension OptimizelyUserContext {
+    
+    /// Sets the forced decision (variation key) for a given flag and an optional rule.
+    /// - Parameters:
+    ///   - flagKey: A flag key.
+    ///   - ruleKey: An experiment or delivery rule key (optional).
+    ///   - variationKey: A variation key.
+    /// - Returns: true if the forced decision has been set successfully.
+    public func setForcedDecision(flagKey: String,
+                                  ruleKey: String? = nil,
+                                  variationKey: String) -> Bool {
+        
+        guard optimizely?.config != nil else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return false
+        }
+        
+        if let index = forcedDecisions.firstIndex(where: { $0.flagKey == flagKey && $0.ruleKey == ruleKey }) {
+            forcedDecisions[index].variationKey = variationKey
+        } else {
+            forcedDecisions.append(ForcedDecision(flagKey: flagKey, ruleKey: ruleKey, variationKey: variationKey))
+        }
+        
+        return true
+    }
+    
+    /// Returns the forced decision for a given flag and an optional rule.
+    /// - Parameters:
+    ///   - flagKey: A flag key.
+    ///   - ruleKey: An experiment or delivery rule key (optional).
+    /// - Returns: A variation key or nil if forced decisions are not set for the parameters.
+    public func getForcedDecision(flagKey: String, ruleKey: String? = nil) -> String? {
+        guard optimizely?.config != nil else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return nil
+        }
+        
+        return findForcedDecision(flagKey: flagKey, ruleKey: ruleKey)
```

Comment:
How come this is broken into two parts? It seems like the functionality of findForcedDecision is the serves the purpose of the get function here just with that logic taken out? Would it make sense to just bring that logic into the getForcedDecision and use the get function where we are calling the find function since the get calls the find?

Reply:
The core function is separated out just to add SDK-ready-check only to the public getForcedDecision API.

Reply:
Makes sense, thank you for explaining that!